Crashlytics/Crashlytics/Components/FIRCLSBinaryImage.m

```diff
@@ -343,23 +344,41 @@ static bool FIRCLSBinaryImageFillInImageDetails(FIRCLSBinaryImageDetails* detail
   FIRCLSMachOSection section;
 
 #if CLS_COMPACT_UNWINDING_SUPPORTED
+#if !defined(TARGET_OS_XR) || !TARGET_OS_XR
   if (FIRCLSBinaryImageMachOSliceInitSectionByName(&details->slice, SEG_TEXT, "__unwind_info",
                                                    &section)) {
     details->node.unwindInfo = (void*)(section.addr + details->vmaddr_slide);
   }
+#else
+  unsigned long unwindInfoSize;
+  details->node.unwindInfo = (void*)getsectiondata(details->slice.startAddress, "__TEXT",
+                                                   "__unwind_info", &unwindInfoSize);
+#endif
 #endif
 
 #if CLS_DWARF_UNWINDING_SUPPORTED
+#if !defined(TARGET_OS_XR) || !TARGET_OS_XR
   if (FIRCLSBinaryImageMachOSliceInitSectionByName(&details->slice, SEG_TEXT, "__eh_frame",
                                                    &section)) {
     details->node.ehFrame = (void*)(section.addr + details->vmaddr_slide);
   }
+#else
+  unsigned long ehFrameSize;
+  details->node.ehFrame =
+      (void*)getsectiondata(details->slice.startAddress, "__TEXT", "__eh_frame", &ehFrameSize);
+#endif
 #endif
 
+#if !defined(TARGET_OS_XR) || !TARGET_OS_XR
```

Comment:
I forget if we talked about this - is it possible for us to define this as a macro internally?

Eg. update https://github.com/firebase/firebase-ios-sdk/blob/master/Crashlytics/Crashlytics/Helpers/FIRCLSDefines.h


`#define CLS_TARGET_OS_XR (defined(TARGET_OS_XR) && TARGET_OS_XR)`

And then use `if !CLS_TARGET_OS_XR` here

Reply:
`#define CLS_TARGET_OS_XR (defined(TARGET_OS_XR) && TARGET_OS_XR)` gives warning for Macro expansion producing 'defined' has undefined behavior. So I changed to instead:
```
#if defined(TARGET_OS_XR) && TARGET_OS_XR
#define CLS_TARGET_OS_XR 1
#else
#define CLS_TARGET_OS_XR 0
#endif
```

Reply:
Thanks, Themis and Sam! I think the `CLS_TARGET_OS_XR` macro makes this more readable.